File: backend/B2BApplication/Models/LignePanier.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace B2BApplication.Models
{
    public class LignePanier
    {
        [Key]
        public int id { get; set; }
        public int articleId { get; set; }
        public Article article { get; set; }
        public int ligneQuantite { get; set; }
        public DateTime dateCreate { get; set; } = DateTime.Now;
        public User userCreate { get; set; }
        [ForeignKey("userCreate")]
        public string userId { get; set; }

    }
}

File: frontend/src/app/models/LignePanier.ts
import { Article } from "./article"
import { User } from "./user"

export class LignePanier{
    id:number
    articleId:number
    ligneQuantite:number
    username:string
    constructor(articleId:number,ligneQuantite:number,username:string){
        this.id=0
this.articleId=articleId
this.ligneQuantite=ligneQuantite
this.username=username
    }
}